### PR TITLE
Journal App front end submission

### PR DESCRIPTION
**Pull request by Somanath1786**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,15 +36,15 @@ By the end of this lesson. You should be able to set up two separate servers tha
 
 * **Question:** What error do you get? Why?
 
-* **Your Answer:** 
+* **Your Answer:**  we get a cors error , because we havent enabled cors on our server
```
Comment (MadEste) [inline README.md:39]: 👍 Correct, we are trying to access a different port.

File: README.md [hunk 2]
```diff
@@ -96,35 +96,37 @@ By the end of this lesson. You should be able to set up two separate servers tha
 
 * **Question:** What is happening on the first couple of lines of the new file you've created?
 
-* **Your Answer:** 
+* **Your Answer:** In the first line we are pulling out the environment varaible and if it is a development environment we want the URL to be localhost and when we deploy it change it to an actual url,
```
Comment (MadEste) [inline README.md:99]: 👍

File: README.md [hunk 3]
```diff
@@ -96,35 +96,37 @@ By the end of this lesson. You should be able to set up two separate servers tha
...
 * **Question:** What changes on the page after you successfully login? Why?
 
-* **Your Answer:**
+* **Your Answer:** Once we set the state, we show a bunch of navigation links
 
 * **Question:** What happens if you enter in the incorrect information? What _should_ happen?
 
-* **Your Answer:**
+* **Your Answer:** we throw an exception and we need to add a try catch block to handle it and give info
+to the end user that they have entered invalid creds
```
Comment (MadEste) [inline README.md:129]: 👍

File: README.md [hunk 4]
```diff
@@ -141,23 +143,23 @@ By the end of this lesson. You should be able to set up two separate servers tha
...
+* **Your Answer:** We want to get the token and update the state and render the navigation links and be logged in if there is a token. This code will run once every time the react component mounts. This way when we user refreshes after loggin in the state is updated correctly and the user continues to remain in the same state as they were before logging in
 
 ---
 
 - [ ] Now when you refresh the page, it looks as though you are logged in. Next, try clicking the logout button.
 
 * **Question:** When you click "Logout", nothing happens unless you refresh the page. Why not?
 
-* **Your Answer:**
+* **Your Answer:** When we click log out at this point of time, we have removed the token, but we havent changed the state and hence we continue to render as if we have a user id
```
Comment (MadEste) [inline README.md:154]: 👍

File: README.md [hunk 5]
```diff
@@ -242,7 +244,10 @@ By the end of this lesson. You should be able to set up two separate servers tha
...
 
-* **Question:**
+* **Question:** The reason the number of posts did not change is that when we deleted the post, we deleted it from the database. However this does not mean that the front end state gets automated automatically. The state variable in the front end still contains stale data and hence refelcts the old number.
+To get around this issue we can follow one of the two approaches
+1. Whenever we update the database, update the state in the front end as well. This is not the most ideal one as there can be conflicts between the server and the client
+2. The better option is to repull the data from the server. If we have a lot of data then we do a lot of network I/O, hence the most optimal thing to do here is to do a GET call for whatever info is needed to be fetched as a part of update. For example, whenever we update a post, we can get all the info that is needed to render the page that we will end up with when we finish the action. So it is better to fect the data from the server, but what data and how much you fetch depends on the amount of data that is being pulled
```
Comment (MadEste) [inline README.md:250]: Great answer and very important to consider on bigger projects!

File: src/components/posts/List/List.js [hunk 6]
```diff
@@ -3,22 +3,51 @@ import React from 'react'
 import Actions from './List.Actions'
 
 export default ({ destroyPost, user }) => {
+  // I have used two different two slightly different variants of conditional rendering for question 1 and question 2 just to try out both approaches
```
Comment (MadEste) [inline src/components/posts/List/List.js:6]: 👍

File: src/components/posts/List/List.js [hunk 7]
```diff
@@ -3,22 +3,51 @@ import React from 'react'
...
+
+  // Question 2 : If there is no emotion for a post, hide the associated message on each post.
+  // Check the posts if emotion is populated, if so return the emotion as a blockquote, esle return nothing, thereby
+  // conditionally rendering that field
+  const displayEmotion = (post) => {
+    if (post.emotion)
```
Comment (MadEste) [inline src/components/posts/List/List.js:12]: may also want to test for an empty string

File: src/components/posts/Container.js [hunk 8]
```diff
@@ -4,27 +4,43 @@ import { Route } from 'react-router-dom'
...
   createPost (post) {
     console.log('Submitting Post:', post)
   }
 
-  destroyPost (post) {
-    console.log('Destroying Post:', post)
+  async destroyPost (userId, postId) {
+    //console.log('Destroying Post:', postId)
+    const response = await posts.deletePosts(userId, postId)
+    console.log(response);
```
Comment (MadEste) [inline src/components/posts/Container.js:29]: Be sure to update state to reflect changes
